rfcs/0053-pull-request-workflow.md

```diff
@@ -0,0 +1,93 @@
+---
+feature: pull-request workflow
+start-date: 2019-09-28
+author: Ingolf Wagner (@mrVanDalo)
+co-authors: lassulus (@lassulus)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: 
+---
+
+# Summary
+[summary]: #summary
+
+Pull-Request on GitHub are the main way we improve our code base.
+This Document should settle everything that needs to be done to bring code upstream.
+
+# Motivation
+[motivation]: #motivation
+
+Eliminate questions from contributors and maintainers
+about what should be done next and who should do it.
+This is not a new approach, it is more a settlement
+on how we do it now.
+This RFC is based on my experiences and talks I had
+with other contributors and maintainers.
+
+# Detailed design
+[design]: #detailed-design
+
+Define all steps of a Pull-Request.
+Use Roles to define responsibilities in every step.
+
+## Roles
+[roles]: #roles
+
+Everybody involved in the process of contributing has one or multiple
+of the following roles
+
+* Contributor
+* Bot
+* Reviewer
+* Maintainer (has merge privileges)
+
+The responsibilities of theses roles are defined in the rest of this RFC.
+
+## States of a Pull-Request
+[state]:#states
+
+This diagram defines all states of a Pull-Request,
+and their transitions to other states.
+
+![pull-request state](0053-pull-request-workflow/pull-request-states.svg)
+
+## Responsibilities and Actions
+[responsibilities]:#responsibilities
+
+The responsibilities of every role is defined by the following diagram:
+
+![pull-request activity](0053-pull-request-workflow/pull-request-roles.svg)
+
+## About Pull-Request
+
+### Packages
+
+* contributor must decide if a Backport is necessary
+* after the Pull-Request to master is merged the Backport Pull-Request is created
+* Backport Pull-Requests must be linked to the original Pull-Requests in master
```

Comment:
Is there any change to tell GitHub to add the [`9.needs: port to stable`](https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+is%3Apr+label%3A%229.needs%3A+port+to+stable%22) in case of a backport by an unprivileged contributor?

Also, what about "manual" backports (commiters cherry-picking patches directly to `release-yy.mm`)?

Reply:
> Is there any change to tell GitHub to add the `9.needs: port to stable` in case of a backport by an unprivileged contributor?

I've seen this tag, but I did not know who is setting that in which situation. This is also one reason why I created this RFC. What are the rules behind this tag? There is also a `8: has port to stable`. 

I would love to put the meaning and the ruling behind theses tags in this RFC.

Reply:
> Also, what about "manual" backports (commiters cherry-picking patches directly to `release-yy.mm`)?

I guess by "manual" you mean without a pull-request. This is something I don't know of at all. The only thing I heard from people is that they don't like it to be done.

Reply:
> may include in parentheses (using `git cherry-pick -x`)

I will add this, but I had more the pull-request-id in mind.